### DISCUSSION TITLE
Testing different CARMA bin number / sizes

### DISCUSSION THREAD

**Discussion by tilmes** [NCAR/CARMA_dev#27] [trop_strat CARMA model]
Based on the setup of the carma_trop_strat model (current version 12 of the development code), we discussed potential shortcomings in the model setup. This Discussion is about setting up new test simulations that explore different bin sizes and numbers. 

We discussed to start with the WACCM-MA model, that is nudged to MERRA data. In the original case, CARMA is potentially not covering the largest sizes properly. The size distribution shows that we may overestimate the number of the largest bin. The initial idea is to increase the mixed and pure group by 2 bins.
<img width="637" alt="image" src="https://user-images.githubusercontent.com/22222754/229848383-2d146302-84d7-407f-8cf4-0f565f7d97c4.png">

The current code base we can start from is:
/glade/work/fvitt/camdev/carma_trop_strat12/src/physics/carma/models/trop_strat_soa1/carma_model_mod.F90

The first test run is a clone from this case. It would start in 1990, to have 1 year of spinup before Mt Pinatubo. 
/glade/p/cesm/chwg_dev/tilmes/cases/carma/FWmaCARMAnudged.f19_f19_mg17.carma_trop_strat.fixall.1990_2000.region.5TgS.Zhao
However, please change the reference date to 1990-01-01.
You  can use these restart files and copy them in your  run-directory after setting up the run:
/glade/scratch/tilmes/archive/FWmaCARMAnudged.f19_f19_mg17.carma_trop_strat.fixall.1987spinup/rest/1990-01-01-00000

In order to change the bin sizes, we need 2 modifications
After setting up your clone, you need to copy this file: /glade/work/fvitt/camdev/carma_trop_strat12/src/physics/carma/models/trop_strat_soa1/carma_model_mod.F90
into your <Casedirectory>/SourceMods/src.cam
There you can modify this file:
Change: NBIN to NBIN = 22
integer, public, parameter      :: NBIN     = 20              !! Number of particle bins
and update these numbers (at least needed for the mixed aerosol)
real(kind=f), public, parameter     :: rmin_PRSUL     = 3.43e-8_f  ! minimum radius (cm)
  real(kind=f), public, parameter     :: vmrat_PRSUL    = 3.67_f     ! volume ratio
  real(kind=f), public, parameter     :: rmin_MXAER     = 5e-6_f     ! minimum radius (cm)
  real(kind=f), public, parameter     :: vmrat_MXAER    = 2.2588_f    !2.4610_f        ! volume ratio

@zywshoon Could  you please remind us, what numbers  you had used for  the MXAER values?

To check the bins, please use Chuck's ncl scripts he sent last week!

The second modification is in your user_nl_cam:
After building your  run, you will find a CaseDocs/atm_in file in your case directory. 
Please copy the following part in your user_nl_cam file and modify:
 &rad_cnst_nl
 bin_defs    .. and all the lines that come after until:
'N:PRSULF20:N:CLDPRSULF20:sulfate:shell:/glade/p/cesmdata/cseg/inputdata/atm/cam/physprops/sulfate_rrtmg_carma_c080918.nc'

 rad_climate            = 'A:Q:H2O', 'N:O2:O2',...
and everything until
'B:PRSUL19:/glade/p/cesmdata/cseg/inputdata/atm/cam/chem/carma/rrtmg/aerosol_cld_SULF19_rrtmg.nc', 'B:PRSUL20:/glade/p/cesmdata/cseg/inputdata/atm/cam/chem/carma/rrtmg/aerosol_cld_SULF20_rrtmg.nc'

Then modify the list and add the two extra bins for both MXAER and PRSUL.
I can share my ncl programs to plot the comparisons to the observations, or do the comparisons.

**Comment by ilaria-quaglia** [comment]
@tilmes  can you share the ncl program you use for the comparison? thanks!

**Reply by tilmes** [reply]
@ilaria-quaglia Have you done a run over Mt Pinatubo or what year?
I have some different code from Yunqian, I can share.
Simone

**Reply by ilaria-quaglia** [reply]
I submitted the simulation for 1 year from 1990

**Comment by tilmes** [comment]
Great! It would be also very helpful if  you  could make a Github "Issue" entry with your simulation, case directory, and where the output will go, and what you did.

**Comment by tilmes** [comment]
@cbardeen could you please again add the ncl program location to see the CARMA bin distribution? I am planning to looking into this some more in the next few days.
